adam_core/coordinates/transform.py

```diff
@@ -1434,13 +1424,28 @@ def transform_coordinates(
         # If we assumed the coordinates lie on a unit sphere and the
         # rho and vrho values were assumed then make sure the output coordinates
         # and covariances are set back to NaN values and masked
-        if set_rho_nan:
-            coords_out._values[:, 0] = COORD_FILL_VALUE
-            coords_out._values[:, 0].mask = 1
 
-        if set_vrho_nan:
-            coords_out._values[:, 3] = COORD_FILL_VALUE
-            coords_out._values[:, 3].mask = 1
+        if set_rho_nan or set_vrho_nan:
+            values_out = coords_out.values
+
+            if set_rho_nan:
+                values_out[:, 0] = np.NaN
+
+            if set_vrho_nan:
+                values_out[:, 3] = np.NaN
+
+            coords = SphericalCoordinates.from_kwargs(
+                rho=values_out[:, 0],
+                lat=values_out[:, 1],
+                lon=values_out[:, 2],
+                vrho=values_out[:, 3],
+                vlat=values_out[:, 4],
+                vlon=values_out[:, 5],
+                covariances=coords.covariances,
```

Comment:
The covariances aren't getting masked with nulls here.

This whole behavior - implicitly casting to a unit sphere - seems error prone, to me. If you converted SphericalCoordinates to Keplerian, for example, you'd get an `a` and a `p` etc that are dependent on this imputed value, and I'm not sure what the covariances would mean at all.

While it's a little bit more convenient to automatically do this, the implications seem too dramatic. I think this makes me _more_ certain that users should call some sort of `coords.to_unit_sphere(self)` explicitly if they want this behavior, rather than putting it inside of this function.

Reply:
I agree and I like your suggestion. I've added a `.to_unit_sphere()`.